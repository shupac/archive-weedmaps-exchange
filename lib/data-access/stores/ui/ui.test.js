import { getSnapshot } from 'mobx-state-tree';
import UiStore from './';

const mockUiStore = {
  notifications: [
    {
      title: 'Toast Title One',
      body: 'this is the toast body',
      id: '1234',
    },
    {
      title: 'Toast Title Two',
      body: 'this is the toast body',
      id: '5678',
    },
  ],
  notifyToast: jest.fn(),
};

describe('UI Store', () => {
  it('can create an instance of a model', () => {
    const uiStore = UiStore.create(mockUiStore);
    expect(getSnapshot(uiStore)).toMatchSnapshot();
  });

  it('handles adding a toast alert', () => {
    const uiStore = UiStore.create(mockUiStore);
    uiStore.notifyToast({
      title: 'toast title',
      body: 'toast body',
    });
    expect(uiStore.notifications.length).toEqual(3);
  });

  it('handles dismissing a toast alert', () => {
    const uiStore = UiStore.create(mockUiStore);
    uiStore.dismissToast('1234');
    expect(uiStore.notifications.length).toEqual(1);
  });
});
