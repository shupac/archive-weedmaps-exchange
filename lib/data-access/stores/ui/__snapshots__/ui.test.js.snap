// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Store can create an instance of a model 1`] = `
Object {
  "isLoading": false,
  "notifications": Array [
    Object {
      "body": "this is the toast body",
      "id": "1234",
      "title": "Toast Title One",
    },
    Object {
      "body": "this is the toast body",
      "id": "5678",
      "title": "Toast Title Two",
    },
  ],
}
`;
